projects/cds/src/merchandising/facade/cds-merchandising-user-context.service.ts

```diff
@@ -27,20 +29,29 @@ export class CdsMerchandisingUserContextService {
   constructor(
     private routingService: RoutingService,
     private productSearchService: ProductSearchService,
-    private converterService: ConverterService
+    private converterService: ConverterService,
+    private profileTagEventService: ProfileTagEventService
   ) {}
 
   getUserContext(): Observable<MerchandisingUserContext> {
-    return merge(
-      this.getCategoryAndFacetContext(),
-      this.getProductNavigationContext()
+    return combineLatest([
+      this.getConsentReferenceContext(),
+      merge(
+        this.getCategoryAndFacetContext(),
+        this.getProductNavigationContext()
+      ),
+    ]).pipe(
+      map(([consentReferenceContext, userContext]) => ({
+        ...consentReferenceContext,
+        ...userContext,
+      }))
     );
   }
 
   private getCategoryAndFacetContext(): Observable<MerchandisingUserContext> {
     return combineLatest([
       this.getCategoryNavigationContext(),
-      this.getFacetsContext(),
+      this.getFacetsContext().pipe(startWith({ facets: undefined })),
```

Comment:
I guess that you had to add `startWith` because `combineLatest` wasn't firing in some case?

Reply:
That's correct, I noticed that on a homepage hit there was nothing output from this stream

Reply:
This kind of defeats the purpose of `isFacetPage()` check that's being done in `getFacetsContext()` method.

As Spartacus uses the ngrx store to store search results (and selecting state slices from this store will always emit something), the idea was to prevent emission unless the user is on the facet-supported page.

Looking at the whole logic, I understand why this had to be added - because of `combineLatest` in the `getUserContext`. However, now that we are switching to sending `consent-reference` header in an interceptor, I think that this `startWith` can safely be removed.

Reply:
I've not been able to remove the combineLatest using interceptor approach, so any ideas would be appreciated

Reply:
I think he was referring to the `startWith` and not remove the `combineLatest` specifically. I think Nikola is trying to say once the interceptor approach is good with all streams properly emitting, then we can safely remove `startWith` from `this.getFacetsContext`, which forces it to emit a value.

Reply:
Unfortunately, on the homepage, if there is no facets, and no navigation context, then the combineLatest does nothing. So, if the consent-reference gets created, the content doesn't get refreshed